kubernetes/utils/create_from_yaml.py

```diff
@@ -23,51 +23,105 @@
 from kubernetes import client
 
 
-def create_from_yaml(k8s_client, yaml_file, verbose=False, **kwargs):
+def create_from_yaml(
+        k8s_client,
+        yaml_file,
+        verbose=False,
+        output_list=False,
+        **kwargs):
     """
     Perform an action from a yaml file. Pass True for verbose to
     print confirmation information.
     Input:
     yaml_file: string. Contains the path to yaml file.
     k8s_cline: an ApiClient object, initialized with the client args.
+    verbose: If True, print confirmation from the create action. Default is False.
+    output_list: compatibility option with v8.0.0. Default is False.
+        Function returns a single api object when there is only one when set False.
+        Does not affect when multiple objects are generated.
 
-    Available parameters for performing the subsequent action:
+    Returns:
+    An k8s api object or list of apis objects created from YAML.
+    When a single object is generated, return type is dependent on output_list.
+
+    Available parameters for creating <kind>:
     :param async_req bool
     :param bool include_uninitialized: If true, partially initialized resources are included in the response.
     :param str pretty: If 'true', then the output is pretty printed.
     :param str dry_run: When present, indicates that modifications should not be persisted. An invalid or unrecognized dryRun directive will result in an error response and no further processing of the request. Valid values are: - All: all dry run stages will be processed
     """
 
+    k8s_api_all = []
     with open(path.abspath(yaml_file)) as f:
```

Comment:
`path.abspath(yaml_file) `
will crash on string input while `yaml.load(str)` can load the yaml from a string.
very useful if you need to change the yaml dynamically

ill suggest to support yaml_file as string and a file

Reply:
I don’t think this is necessary. The feature requested is to replicate kubectl create -f which of course would not support strings. YAML files can be load as a dictionary and passed as the body parameter when the user calls the corresponding APIs. I don’t see the benefit of changing a YAML on the fly - we’re not doing kubectl apply -f here.

If you want this feature to parse as a string, please file a separate issue.